rust/core-lib/src/view.rs

```diff
@@ -220,6 +220,26 @@ impl View {
         self.set_selection_raw(text, selection);
     }
 
+    /// Selects a specific range (eg. when the user performs SHIFT + click).
+    pub fn select_range(&mut self, text: &Rope, offset: usize) {
+      if !self.is_point_in_selection(offset) {
+        let sel = {
+          let (last, rest) = self.sel_regions().split_last().unwrap();
+          let mut sel = Selection::new();
+          for &region in rest {
+            sel.add_region(region);
+          }
+          // TODO: small nit, merged region should be backward if end < start.
+          // This could be done by explicitly overriding, or by tweaking the
+          // merge logic.
+          sel.add_region(SelRegion::new(last.start, offset));
+          sel
+        };
+        self.set_selection(text, sel);
```

Comment:
This is moved over from the deleted `do_click` fn in `editor.rs`. In any case the rationale is that `View::set_selection` has a few side effects (well: _effects_) such as modifying the view's scroll position if necessary. 

There are definitely other ways of handling this, but some sort of copying is currently necessary because you can't manipulate the selection in place.